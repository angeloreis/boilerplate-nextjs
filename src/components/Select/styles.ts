import styled from 'styled-components'

export const Wrapper = styled.select`
  border-radius: 5px;
  padding: 2rem 1rem;
  font-size: 1.8rem;
  min-width: 128px;
  width: 100%;
  max-width: 512px;
`
