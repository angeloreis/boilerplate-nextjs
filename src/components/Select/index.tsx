import * as S from './styles'

export interface OptionsProps {
  value: string | number
  description: string
  defaultValue?: boolean
}

interface SelectProps {
  data: OptionsProps[]
  placeholder?: string
  onChangeHandle?: () => Promise<void>
}

const Select = ({
  data,
  placeholder = 'Escolha uma opção...',
  onChangeHandle
}: SelectProps) => {
  return (
    <>
      {data && data.length > 0 ? (
        <S.Wrapper onChange={onChangeHandle} placeholder={placeholder}>
          <option value="initial">{placeholder}</option>
          {data.map((option, index) => (
            <option
              key={index}
              value={option.value}
              defaultValue={option.defaultValue ? option.value : undefined}
            >
              {option.description}
            </option>
          ))}
        </S.Wrapper>
      ) : (
        <h1>Nenhum dado informado</h1>
      )}
    </>
  )
}

export default Select
