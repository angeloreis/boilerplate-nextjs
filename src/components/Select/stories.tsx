import { Story, Meta } from '@storybook/react'
import Select from '.'

export default {
  title: 'Select',
  component: Select,
  args: {
    data: [
      {
        value: 'Audi V3',
        description: 'Audio V3 1999'
      },
      {
        value: 'Audi V4',
        description: 'Audio V4 2000'
      },
      {
        value: 'Audi V4.1',
        description: 'Audio V3 2000/2001'
      },
      {
        value: 'Audi V5',
        description: 'Audio V3 2001',
        defaultValue: true
      }
    ]
  },
  argTypes: {
    data: {
      required: true,
      name: 'Data of options',
      control: { type: 'array' }
    }
  }
} as Meta

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const Template = (args: any) => {
  return <Select {...args} />
}

export const Default: Story = Template.bind({})
