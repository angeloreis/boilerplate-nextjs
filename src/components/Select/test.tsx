import { render, screen } from '@testing-library/react'

import Select from '.'

describe('<Select />', () => {
  it('should render the heading', () => {
    const data = [
      {
        value: 'Audi V3',
        description: 'Audi V3 1999'
      },
      {
        value: 'Audi V4',
        description: 'Audi V4 2000'
      },
      {
        value: 'Audi V4.1',
        description: 'Audi V3 2000/2001'
      },
      {
        value: 'Audi V5',
        description: 'Audi V3 2001',
        defaultValue: true
      }
    ]
    const { container } = render(<Select data={data} />)
    const componentSelect = screen.getByRole('option', { name: /uma/i })

    expect(componentSelect).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
