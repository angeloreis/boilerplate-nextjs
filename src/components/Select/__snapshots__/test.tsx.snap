// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render the heading 1`] = `
.c0 {
  border-radius: 5px;
  padding: 2rem 1rem;
  font-size: 1.8rem;
  min-width: 128px;
  width: 100%;
  max-width: 512px;
}

<select
  class="c0"
  placeholder="Escolha uma opção..."
>
  <option
    value="initial"
  >
    Escolha uma opção...
  </option>
  <option
    value="Audi V3"
  >
    Audi V3 1999
  </option>
  <option
    value="Audi V4"
  >
    Audi V4 2000
  </option>
  <option
    value="Audi V4.1"
  >
    Audi V3 2000/2001
  </option>
  <option
    value="Audi V5"
  >
    Audi V3 2001
  </option>
</select>
`
